src/targeting/youtube-ima.ts

```diff
@@ -20,14 +21,34 @@ const encodeCustomParams = (
 	return encodedParams;
 };
 
+const mergeCustomParamsWithTargeting = (
+	customParams: CustomParams,
+	consentState: ConsentState,
+) => {
+	try {
+		const pageTargeting = buildPageTargeting(consentState, false);
+		const mergedCustomParams = { ...customParams, ...pageTargeting };
+		return mergedCustomParams;
+	} catch (e) {
+		/**
+		 * Defensive error handling in case YoutubeAtom is used in an
+		 * environment where guardian.config, cookies, localstorage etc
+		 * are not available
+		 */
+		log('commercial', 'Error building YouTube IMA custom params', e);
+	}
+	return customParams;
```

Comment:
Does the following make it clearer that it's specifically `buildPageTargeting(...)` that can throw an exception?

```
try {
		const pageTargeting = buildPageTargeting(consentState, false);
	} catch (e) {
		/**
		 * Defensive error handling in case YoutubeAtom is used in an
		 * environment where guardian.config, cookies, localstorage etc
		 * are not available
		 */
		log('commercial', 'Error building YouTube IMA custom params', e);
		return customParams;
	}

	const mergedCustomParams = { ...customParams, ...pageTargeting };
	return mergedCustomParams;
};
```

Reply:
That's much clearer. Thanks!